walkthroughs/week-2-server/examples/page-view-counter/src/main/java/com/google/sps/servlets/PageViewServlet.java

```diff
@@ -24,14 +25,14 @@
 @WebServlet("/page-views")
 public class PageViewServlet extends HttpServlet {
 
-  private int pageViews = 0;
+  private AtomicInteger pageViews = new AtomicInteger();
```

Comment:
Most participants will not be familiar with threading, so we've disabled multithreading in all of these projects (see the `threadsafe` property in the `appengine-web.xml` files). So I'd vote to keep this as-is, even if it's not thread-safe.

If you're interested in introducing concurrency, this would make a good lecture session!

Reply:
The possibility that AppEngine might serialize requests had never even occurred to me. /shudder :P

My documentation change about "concurrent requests" is definitely wrong then, too.

That also explains why the SubtractionGame didn't crash with concurrent array mutation errors in my load test. I really wanted to look at making this use per-session state instead of a single global game.. but that's also just distracting in this super busy week of learning.

Reply:
Yeah, exactly. There are so many things we could introduce, but for the sake of time I've tried to boil it down to the bare minimum required to get something interesting working. From there I think there's plenty of room for "extra credit" sessions, where e.g. somebody talks about concurrency for an hour over GVC. I think that would be a cool way to show different advanced topics without overwhelming people who are new to Java.